lib/allflex/product_mapper/allflex_product.rb

```diff
@@ -231,7 +238,8 @@ def marked?
       # the id will either be defined in the maufacturing definition,
       # or it will need to be build algorithmically from the manufacturing definition and the components
       def generate_id
-        return Failure("No ordered_id_components") unless ordered_id_components
+        return Failure(ordered_id_components: [{ code: 'No ordered product id components passed in', errors: { message: 'Error in generate_id method in allflex product' }, context: {} }]) unless ordered_id_components
```

Comment:
Where is the test for this change?

```suggestion
        return Failure(ordered_id_components: [{ code: 'No ordered product id components passed in', errors: [{ message: 'Error in generate_id method in allflex product' }], context: {} }]) unless ordered_id_components
```

Reply:
Fixed the errors structure within the app too

Reply:
Done a test for this one